jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TaskCommon.java

```diff
@@ -67,14 +68,32 @@ static void disableHttpLogging() {
     java.util.logging.Logger.getLogger(HttpTransport.class.getName()).setLevel(Level.OFF);
   }
 
+  @Deprecated
+  static void checkDeprecatedUsage(JibExtension jibExtension, Logger logger) {
+    if (jibExtension.extraDirectoryConfigured
+        || System.getProperty(PropertyNames.EXTRA_DIRECTORY_PATH) != null
+        || System.getProperty(PropertyNames.EXTRA_DIRECTORY_PERMISSIONS) != null) {
+      logger.warn(
+          "'jib.extraDirectory', 'jib.extraDirectory.path', and 'jib.extraDirectory.permissions' "
+              + "are deprecated; use 'jib.extraDirectories', 'jib.extraDirectories.paths' and "
+              + "'jib.extraDirectories.permissions'");
+
+      if (jibExtension.extraDirectoriesConfigured
+          || System.getProperty(PropertyNames.EXTRA_DIRECTORIES_PATHS) != null
+          || System.getProperty(PropertyNames.EXTRA_DIRECTORIES_PERMISSIONS) != null) {
+        throw new IllegalArgumentException(
+            "You cannot configure both 'jib.extraDirectory.path' and 'jib.extraDirectories.paths'");
```

Comment:
This message about `jib.extraDirectory.path`/`jib.extraDirectories.paths` will show even if you're only using `jib.extraDirectory`/`jib.extraDirectories`, I think. Might be better to just say "You cannot configure both 'jib.extraDirectory' and 'jib.extraDirectories'".

Reply:
Good point. I considered it, and there's a reason that I ended up using `.path(s)`.

There were some limitations, and I couldn't find a way whether a user configured `jib.extraDirectory` or `jib.extraDirectory.path` (you can get the sense in https://github.com/GoogleContainerTools/jib/pull/1671#discussion_r280123597). If the user used `jib.extraDirectory.(path)` and saw the message without `.path(s)`, the user may just try change "Directory" to "Directories" and run into another problem, because we won't support the shortcut `jib.extraDirectories` anymore and `jib.extraDirectories.path` won't work either (`path` should be pluralized). So I think for less friction in migration, I thought it'd be better to mention the `path(s)` part than not mentioning it.

Reply:
Oh, I didn't realize we weren't support the `jib.extraDirectories` shortcut anymore. In that case this makes sense.